src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -1240,6 +1218,23 @@ class VariantSampleMatrix[T](val hc: HailContext, val metadata: VariantMetadata,
     filterSamples(p)
   }
 
+  def filterSamplesTable(table: KeyTable, keep: Boolean): VariantSampleMatrix[T] = {
+    table.keyFields.map(_.typ) match {
+      case Array(TString) =>
+        val sampleSet = table.keyedRDD()
```

Comment:
should I broadcast the sample IDs in the VDS and filter to those? Seemed a bit unnecessary but am happy to do that.